ingestion/src/kldata.rs

```diff
@@ -0,0 +1,279 @@
+
+    let mut data = Vec::with_capacity(chunk.observations.len());
+
+    for in_datum in chunk.observations {
+        // get the conversion first, so we avoid wasting a tsid if it doesn't exist
+        let (element_id, param_id) =
+            param_conversions
+                .get(&in_datum.id.param_code)
+                .ok_or_else(|| {
+                    Error::Parse(format!(
+                        "no element_id found for param_code {}",
+                        in_datum.id.param_code
+                    ))
+                })?;
+
+        if !timeseries_is_open(
+            permit_table.clone(),
+            chunk.station_id,
+            chunk.type_id,
+            param_id.to_owned(),
+            in_datum.id.sensor_and_level,
+            in_datum.timestamp,
+        )? {
+            continue;
+        }
+
+        let transaction = conn.transaction().await?;
+
+        let (sensor, lvl) = in_datum
+            .id
+            .sensor_and_level
+            .map(|both| (Some(both.0), Some(both.1)))
+            .unwrap_or((None, None));
+
+        let obsinn_label_result = transaction
+            .query_opt(
+                &query_get_obsinn,
+                &[
+                    &chunk.station_id,
+                    &chunk.type_id,
+                    &in_datum.id.param_code,
+                    &lvl,
+                    &sensor,
+                ],
+            )
+            .await?;
+
+        let timeseries_id: i32 = match obsinn_label_result {
+            Some(row) => row.get(0),
+            None => {
+                // create new timeseries
+                let timeseries_id = transaction
+                    .query_one(
+                        "INSERT INTO public.timeseries (fromtime) VALUES ($1) RETURNING id",
+                        &[&in_datum.timestamp],
+                    )
+                    .await?
+                    .get(0);
+
+                // create obsinn label
+                transaction
+                    .execute(
+                        "INSERT INTO labels.obsinn \
+                                (timeseries, nationalnummer, type_id, param_code, lvl, sensor) \
+                            VALUES ($1, $2, $3, $4, $5, $6)",
+                        &[
+                            &timeseries_id,
+                            &chunk.station_id,
+                            &chunk.type_id,
+                            &in_datum.id.param_code,
+                            &lvl,
+                            &sensor,
+                        ],
+                    )
+                    .await?;
+
+                // create filter label
+                transaction
+                    .execute(
+                        "INSERT INTO labels.filter \
+                                (timeseries, station_id, element_id, lvl, sensor) \
+                            VALUES ($1, $2, $3, $4, $5)",
+                        &[&timeseries_id, &chunk.station_id, element_id, &lvl, &sensor],
```

Comment:
@ketilt

Reply:
I believe we had a discussion before, and the conclusion was (as is encoded here) that different type_ids will all get their own timeseries (I remember the argument from you and Søren being that the lower res series don't match up to our products because the aggregation is done on the station level, and for some reason I  don't understand, this is seen as desirable). Then a lookup on the filter table will return all these timeseries, and it's up to frost to decide how to use them.

Reply:
On further inspection here, I think you would have an issue that multiple obsinn timeseries_id would map to one filter label since we collapse on type_id. I think its best we avoid this logic here for now. Ketil had logic for timeseries recipes that either needs to be abstracted into the product layer or reimplemented somehow at some point. But this merge request is probably best left cleaner / simpler.

Reply:
> multiple obsinn timeseries_id would map to one filter label since we collapse on type_id

No, that's not what happens. Each gets its own filter label, just a query by frost will return all those filter labels.

I wanted to discard all timeseries except the highest res one, but this was shot down by **you** and Søren. If you want different logic here, i.e deciding to only filter label one of the timeseries, you need to lay out how it will be done, and justify why.

Personally I think this is fine as is, since if frost only wants the highest res timeseries it can just take that one and ignore the rest. I'm pretty sure that's what happens in current Frost and ODA anyway, since I've seen responses that give me multiple resolutions. The only change I would advocate to make is adding type_id to filter, so frost can do that more easily.

Reply:
As with the above, this should be resolved with the change to a met label